src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -101,40 +96,45 @@ function beginWork(workInProgress : Fiber) : ?Fiber {
   }
   switch (workInProgress.tag) {
     case IndeterminateComponent:
-      mountIndeterminateComponent(workInProgress);
-      break;
+      mountIndeterminateComponent(current, workInProgress);
+      return workInProgress.child;
     case FunctionalComponent:
-      updateFunctionalComponent(workInProgress);
-      break;
+      updateFunctionalComponent(current, workInProgress);
+      return workInProgress.child;
     case ClassComponent:
       console.log('class component', workInProgress.input.type.name);
-      break;
+      return workInProgress.child;
     case HostComponent:
-      updateHostComponent(workInProgress);
-      break;
+      updateHostComponent(current, workInProgress);
+      return workInProgress.child;
     case CoroutineHandlerPhase:
       // This is a restart. Reset the tag to the initial phase.
       workInProgress.tag = CoroutineComponent;
       // Intentionally fall through since this is now the same.
     case CoroutineComponent:
-      updateCoroutineComponent(workInProgress);
+      updateCoroutineComponent(current, workInProgress);
       // This doesn't take arbitrary time so we could synchronously just begin
       // eagerly do the work of workInProgress.child as an optimization.
       if (workInProgress.child) {
-        return beginWork(workInProgress.child);
+        return beginWork(
+          workInProgress.child.alternate,
```

Comment:
That means that this child is new. We don't need an alternate until we try to update it. Remember that we only have one copy until we update. If later we try to update it, this get created through `cloneFiber`.